usr/src/uts/i86pc/os/intr.c

```diff
@@ -471,6 +472,7 @@
 #include <sys/hypervisor.h>
 #endif
 
+boolean_t kernel_preempt_enable = B_FALSE;
```

Comment:
Right, we hit that issue with KVM too, and I think even used that same thing as a workaround.  I'm still unsure why preempting the kernel on runrun as well as kprunrun is a long-term correct fix.  Isn't that precisely the thing that makes runrun and kprunrun different?  I seem to recall with KVM the issue was that we never hit a path that checked kprunrun.  Perhaps I'm just really really confused?  I asked @rmustacc and @jclulow on IRC to have opinions.  They're harder to confuse than I am.